common/rpc_service.go

```diff
@@ -216,15 +232,32 @@ func (sm *serviceMap) UnRegister(protocol, serviceId string) error {
 		sm.mutex.RUnlock()
 		return perrors.New("no services for " + protocol)
 	}
-	_, ok = svcs[serviceId]
+	s, ok := svcs[serviceId]
 	if !ok {
 		sm.mutex.RUnlock()
 		return perrors.New("no service for " + serviceId)
 	}
+	svrs, ok := sm.interfaceMap[interfaceName]
+	if !ok {
+		sm.mutex.RUnlock()
+		return perrors.New("no service for " + interfaceName)
+	}
+	index := -1
+	for i, svr := range svrs {
+		if svr == s {
+			index = i
+		}
+	}
 	sm.mutex.RUnlock()
```

Comment:
this code block should be in a lambda func as follows
func() {
lock()
defer unlock()

}()